src/components/datatablesContainer/DataTableSystems/DataTableSystems.js

```diff
@@ -679,28 +678,49 @@ export const DataTableSystems = ({
       setErrorMsgs(validationErrors);
       return false;
     }
+    
     try {
-      const resp = await mpApi
-        .saveSystemsComponents(
+          let resp;
+          let response;
+
+          response = await mpApi
+          .saveComponents(
+          userInput,
+          selectedSystem.locationId,
+          selectedRangeInFirst.componentId
+          )
+          .catch((error) => console.log("saveComponents failed", error));
+
+          resp = await mpApi
+          .saveSystemsComponents(
           userInput,
           selectedSystem.locationId,
           selectedSystem.id,
           selectedRangeInFirst.id
-        )
+          )
         .catch((error) => console.log("saveSystemsComponents failed", error));
-      if (resp.status >= 200 && resp.status < 300) {
+
+        const newErrorMsgs = [];
+        [resp, response].forEach((res) => {
+          if (res?.status < 200 || res?.status >= 300) {
+                newErrorMsgs.push(res);
+              }
+          });
+
+        if (newErrorMsgs.length > 0) {
+            setErrorMsgs(newErrorMsgs.flat());
+            return false;
+        }
+
         setupdateComponentTable(true);
         setUpdateRelatedTables(true);
         setErrorMsgs([]);
         return true;
-      } else {
-        const errorResp = Array.isArray(resp) ? resp : [resp];
-        setErrorMsgs(errorResp);
+
+      } catch (error) {
+        setErrorMsgs(JSON.stringify(error));
```

Comment:
The original code here was correct, `setErrorMsgs` takes an array argument, not a string:
```suggestion
        setErrorMsgs([JSON.stringify(error)]);
```

Reply:
Updated the ReactHook value as an array